flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointStatementITCase.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @AfterEach
+    public void after() throws Exception {
+        connection.close();
+    }
+
+    public static Stream<String> listHiveSqlTests() throws Exception {
+        return listTestSpecInTheSameModule("endpoint");
+    }
+
+    @ParameterizedTest
+    @MethodSource("listHiveSqlTests")
+    public void testHiveSqlStatements(String sqlPath) throws Exception {
+        runTest(sqlPath);
+    }
+
+    @Override
+    protected String runSingleStatement(String sql) throws Exception {
+        Statement statement = connection.createStatement();
+        statement.execute(sql);
+
+        ResultSet resultSet = statement.getResultSet();
+        ResultSetMetaData metaData = resultSet.getMetaData();
+
+        int columnSize = metaData.getColumnCount();
+        List<RowData> rows = new ArrayList<>();
+        DataType type = toStringifiedType(metaData);
+        while (resultSet.next()) {
+            GenericRowData stringifiedRowData = new GenericRowData(columnSize);
+            for (int i = 0; i < columnSize; i++) {
+                Object field = resultSet.getObject(i + 1);
+                // Similar to SIMPLE_ROW_DATA_TO_STRING_CONVERTER
+                if (field != null) {
+                    if (field instanceof Boolean) {
+                        stringifiedRowData.setField(i, field);
+                    } else if (field instanceof byte[]) {
+                        stringifiedRowData.setField(
+                                i,
+                                StringData.fromString(
+                                        new String((byte[]) field, StandardCharsets.UTF_8)));
+                    } else {
+                        stringifiedRowData.setField(i, StringData.fromString(field.toString()));
+                    }
+                }
+            }
+            rows.add(stringifiedRowData);
+        }
+
+        StatementType statementType = StatementType.match(sql);
+
+        return toString(
+                statementType,
+                DataTypeUtils.expandCompositeTypeToSchema(type),
+                SIMPLE_ROW_DATA_TO_STRING_CONVERTER,
+                rows.iterator());
+    }
+
+    @Override
+    protected String stringifyException(Throwable t) {
+        return t.getMessage().trim();
+    }
+
+    @Override
+    protected boolean isStreaming() throws Exception {
+        Field sessHandleField = HiveConnection.class.getDeclaredField("sessHandle");
+        // Set the accessibility as true
+        sessHandleField.setAccessible(true);
+        SessionHandle sessionHandle =
+                ThriftObjectConversions.toSessionHandle(
+                        (TSessionHandle) sessHandleField.get(connection));
+        return Configuration.fromMap(service.getSessionConfig(sessionHandle))
+                .get(ExecutionOptions.RUNTIME_MODE)
+                .equals(RuntimeExecutionMode.STREAMING);
+    }
+
+    @Override
+    protected void initializeSession() throws Exception {
```

Comment:
This method looks very strange to me. I was thinking it's a pre-action for `testHiveSqlStatements` as well. Rename the method to `resetSessionForFlinkSqlStatements` ?  Alternatively, we can separate `testFlinkSqlStatements` and `testHiveSqlStatements` into different test classes.

Reply:
Because many codes are shared. I think it's better to rename the method name.